features/plugin.feature

```diff
@@ -547,3 +548,33 @@ Feature: Manage WordPress plugins
     Then STDOUT should be a table containing rows:
       | name         | title | description                    |
       | db-error.php |       | Custom database error message. |
+
+  @require-wp-4.0
+  Scenario: Validate installed plugin's version.
+    Given a WP installation
+    And I run `wp plugin install hello-dolly`
+    And a wp-content/mu-plugins/test-plugin-update.php file:
+      """
+      <?php
+      /**
+       * Plugin Name: Test Plugin Update
+       * Description: Fakes installed plugin's data to verify plugin version mismatch
+       * Author: WP-CLI tests
+       */
+
+      add_filter( 'site_transient_update_plugins', function( $value ) {
```

Comment:
I'm still not happy with the flow here, there's still a logical gap (you might have an existing `$value->no_update['hello-dolly/hello.php']` and just keep it...).

I suggest rewriting this like so (and hope WP won't return even more nonsense in that filter):
```php
add_filter( 'site_transient_update_plugins', function( $value ) {
    if ( ! is_object( $value ) ) {
        return $value;
    }

    unset( $value->response['hello-dolly/hello.php'] );
    $value->no_update['hello-dolly/hello.php']->new_version = '1.5';

    return $value;
} );
```